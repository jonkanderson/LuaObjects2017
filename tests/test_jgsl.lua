--[=[
--]=]

package.cpath = package.cpath..';../out/lib/?.so'
local constructors = require('jgsl')

local jCdf = constructors.Cdf()
print("N(0,1), x<-2.5, p=", jCdf:gaussianP(-2.5, 1)) -- pnorm(-2.5, 0, 1)
print("N(0,1), p<0.05, x=", jCdf:gaussianPinv(0.05, 1)) -- qnorm(0.05, 0, 1)
print("N(2,10), x<1.7, p=", jCdf:gaussianP(1.7-2, 10)) -- pnorm(1.7, 2, 10)
print("N(2,10), p<0.30, x=", jCdf:gaussianPinv(0.30, 10)+2) -- qnorm(0.30, 2, 10)
print("t_3, x<-1, p=", jCdf:tdistP(-1, 3)) -- pt(-1, 3)
print("t_3, p<0.2, x=", jCdf:tdistPinv(0.2, 3)) -- qt(0.2, 3)
print("X^2(3), x>2.2, p=", 1-jCdf:chisqP(2.2, 3)) -- 1-pchisq(2.2, 3)
print("X^2(3), p<0.95, x=", jCdf:chisqPinv(0.95, 3)) -- qchisq(0.95, 3)

local jTime = constructors.Time()
local epocSec,nanoSec = jTime:getSecAndNsec()
print("Seconds since epoc =", epocSec)
print("nanoSec=", nanoSec)

--local jRand = constructors.Rand(nanoSec)
local jRand = constructors.Rand(992291721)
print("Some random numbers:")
for i=1,5 do
	print(" Integer", jRand:getInteger())
end
for i=1,5 do
	print(" Uniform", jRand:getUniform())
end
print(" N(0,1)", jRand:getGaussian())
print(" N(0,1)", jRand:getGaussian(1))
print(" N(0,100)", jRand:getGaussian(100))
print(" N(0,100)", jRand:getGaussian(100))
print(" X^2(3)", jRand:getChisq(3))
print(" X^2(3)", jRand:getChisq(3))
print(" t_3", jRand:getTdist(3))
print(" t_3", jRand:getTdist(3))

