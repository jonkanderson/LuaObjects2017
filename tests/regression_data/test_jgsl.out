N(0,1), x<-2.5, p=	0.0062096653257761
N(0,1), p<0.05, x=	-1.6448536269515
N(2,10), x<1.7, p=	0.48803352658589
N(2,10), p<0.30, x=	-3.2440051270804
t_3, x<-1, p=	0.19550110947789
t_3, p<0.2, x=	-0.97847231236331
X^2(3), x>2.2, p=	0.53194837121049
X^2(3), p<0.95, x=	7.8147279032512
Seconds since epoc =	1594237286
nanoSec=	217089302
Some random numbers:
 Integer	2006754787
 Integer	1237284904
 Integer	768764046
 Integer	3702312763
 Integer	1742556254
 Uniform	0.82132338755764
 Uniform	0.048091298434883
 Uniform	0.89486798038706
 Uniform	0.30343838711269
 Uniform	0.92416593036614
 N(0,1)	-0.2460394516482
 N(0,1)	-0.72773264089988
 N(0,100)	-24.804335397682
 N(0,100)	-70.943963983186
 X^2(3)	1.926272296447
 X^2(3)	0.84703377508973
 t_3	-2.5021813620126
 t_3	-2.490362991749
